app/models/shipit/task_definition.rb

```diff
@@ -37,9 +38,14 @@ def as_json
         action: action,
         description: description,
         steps: steps,
+        variables: variables.map(&:to_h),
         checklist: checklist,
         allow_concurrency: allow_concurrency?,
       }
     end
+
+    def task_variables(config_variables)
+      config_variables.map(&VariableDefinition.method(:new))
```

Comment:
I personally don't like using the lambda conversion style because it's harder to understand, and longer than `config_variables.map{ |v| VariableDefinition.new(v) }` but...